spinalcordtoolbox/image.py

```diff
@@ -411,31 +411,21 @@ def loadFromPath(self, path, verbose):
         else:
             logger.debug("Loaded %s orientation %s shape %s", path, self.orientation, self.data.shape)
 
-    def change_shape(self, shape, generate_path=False):
+    def change_shape(self, shape):
         """
         Change data shape (in-place)
 
-        :param generate_path: whether to create a derived path name from the\
-                              original absolutepath (note: while it will generate\
-                              a file suffix, don't expect the suffix but rather\
-                              use the Image's absolutepath.\
-                              If not set, the absolutepath is voided.
-
         This is mostly useful for adding/removing a fourth dimension,
         you probably don't want to use this function.
 
         """
         if shape is not None:
             change_shape(self, shape, self)
-
-        if generate_path and self._path is not None:
-            self._path = add_suffix(self._path, "_shape-{}".format("-".join([str(x) for x in shape])))
         else:
-            # safe option: remove path to avoid overwrites
-            self._path = None
+            raise ValueError("Need to specify a shape!")
```

Comment:
Is this exception necessary? The `shape` parameter has no default value, so this check will only catch someone explicitly passing `None` to `change_shape`.

I think it might be better to let all value errors be caught by `numpy` instead (when `data.reshape` is called later on).

Reply:
Yes you're right.